youtube_dl/extractor/rtvs.py

```diff
@@ -36,12 +43,66 @@ class RTVSIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
+        if url.find('/radio/') != -1:
```

Comment:
`url.split('/')[-2]` is a valid expression, but the test and split can be combined with more assurance that a1, a2 aren't empty:
```
        mobj = re.search(r'(?=.*/radio/).+/([^/]+)/([^/]+)$', url)
        # or if 'radio' shouldn't match the a1/a2 groups
        # mobj = re.search(r'/radio(?:/[^/]+)*/([^/]+)/([^/]+)$', url)
        if mobj:
            # now mobj.groups() is essentially (a1, a2, )
            embed = self._download_webpage(
                'https://www.rtvs.sk/embed/radio/archive/%s/%s' % mobj.groups(),
                video_id)
            ...
```